R/derive_param_tte.R

```diff
@@ -332,27 +304,31 @@ derive_param_tte <- function(dataset = NULL,
   if (create_datetime) {
     date_var <- sym("ADTM")
     start_var <- sym("STARTDTM")
-  }
-  else {
+  } else {
     date_var <- sym("ADT")
     start_var <- sym("STARTDT")
   }
   adsl_vars <- vars(
     !!!subject_keys,
     !!start_var := !!start_date
   )
-  if (!quo_is_null(start_date_imputation_flag)) {
+
+  start_date_imputation_flag <- gsub("(DT|DTM)$", "DTF", as_name(start_date))
+  if (start_date_imputation_flag %in% colnames(dataset_adsl)) {
     adsl_vars <- vars(
       !!!adsl_vars,
-      STARTDTF = !!start_date_imputation_flag
+      STARTDTF = !!sym(start_date_imputation_flag)
     )
   }
-  if (!quo_is_null(start_time_imputation_flag)) {
+
+  start_time_imputation_flag <- gsub("DTM$", "TMF", as_name(start_date))
```

Comment:
this will always evaluate true, i think, because `gsub` will return the original string if it doesn't find the requested regex., and `start_date` is required to run the function. in the tests, this results in a time imputation flag `STARTTMF` that equals the values `TRTSDT`. 

happy to make a change; is it too hacky to just change the if statement to include logic that `as_name(start_date)!=start_time_imputation_flag`?